src/dbnode/storage/index/block.go

```diff
@@ -196,6 +197,7 @@ func newBlockMetrics(s tally.Scope) blockMetrics {
 		queryDocsMatched:       s.Histogram("query-docs-matched", buckets),
 		aggregateSeriesMatched: s.Histogram("aggregate-series-matched", buckets),
 		aggregateDocsMatched:   s.Histogram("aggregate-docs-matched", buckets),
+		reconciledEntry:        s.Counter("reconciled-entry"),
```

Comment:
Perhaps we can rename this to be clear, such as `reconciled-entry-on-query` or similar? 

It might be worth if we change the `ReconciledTryMarkIndexGarbageCollected` method return a similar boolean to add another metric such as `reconciled-entry-on-mark-gc`.